keps/976-plain-pods/README.md

```diff
@@ -427,58 +406,46 @@ spec:
               cpu: 1m
 ```
 
-#### Groups of Pods with the same shape
+#### Groups of Pods created beforehand
 
-When multiple pods belong to the same group and have the same shape, we need to know how many pods
-belong to the group.
+When a group of pods have different shapes, we need to group them into buckets of similar specs in
+order to create a Workload object.
 
-These groups of Pods can be identified when they have:
+To fully identify the group of pods, the pods need the following:
 - the label `kueue.x-k8s.io/pod-group-name`, as a unique identifier for the group. This should
   be a valid CRD name.
-- the annotation `kueue.x-k8s.io/pod-group-count`, the number of pods to expect in the group.
-  This number is interpreted as int32.
+- The annotation `kueue.x-k8s.io/pod-group-total-count` to indicate how many pods to expect in
+  the group.
+
+The Pod reconciler would group the pods into similar buckets by only looking at the fields that are
+relevant to admission, scheduling and/or autoscaling.
+This list might need to be updated for Kubernetes versions that add new fields relevant to 
+scheduling. The list of fields to keep are:
+- In `metadata`: `labels` (ignoring labels with the `kueue.x-k8s.io/` prefix)
+- In `spec`:
+  - In `initContainers` and `containers`: `image`, `requests` and `ports`.
+  - `nodeSelector`
+  - `affinity`
+  - `tolerations`
+  - `runtimeClassName`
+  - `priority`
+  - `preemptionPolicy`
+  - `topologySpreadConstraints`
+  - `overhead`
+  - `volumes`
+
+A sha256 of the reamining Pod spec will be used as a name for a Workload podSet. The count for the
+podSet will be the number of Pods that match the same sha256.
+
+We can only build the Workload object once we observe the number of Pods defined by the
+`kueue.x-k8s.io/pod-group-total-count` annotation.
+
+If Pods with the same `pod-group-name` have different values for the `pod-group-total-count`
+annotation, or if Kueue observes a different amount of Pods than the count, it will not create a
+Workload object and it will emit an event for the Pod indicating the reason.
```

Comment:
What happens if Kueue absorbs a different amount of Pods than the count after creating Workload? The Workload will be deleted?

Reply:
Ah, I see. Thanks for the clarification.